kgo: kongpluginbinding guide (#8108)
Kong/docs.konghq.com: app/_src/gateway-operator/guides/konnect-entities/konnect-plugin-binding.md

title: Managing Plugin Bindings by CRD
---

In this guide you'll learn how to use the `KongPluginBinding` to bind plugins to {{site.konnect_short_name }} entities
from within your Kubernetes cluster.

{% include md/kgo/konnect-entities-prerequisites.md disable_accordian=false version=page.version release=page.release
with-control-plane=true %}

### Introduction of `KongPluginBinding` CRD

The `KongPluginBinding` is the CRD used to manage the binding relationship between plugins and attached {{site.konnect_short_name}} entities, including services, routes, consumers, and consumer groups, or a supported combination of these entities.

> It has two parts for the binding description in its specification: `spec.pluginRef` to refer to a `KongPlugin` resource which contains the plugin name and configuration of the plugin, and `spec.targets` to refer to the entity or combination of entities that the plugin attached to.

Reviewer: ```suggestion
This CRD has two parts for the binding description in its specification: 
* `spec.pluginRef`: Refers to a `KongPlugin` resource which contains the plugin name and configuration of the plugin.
* `spec.targets`: Refers to the entity or combination of entities that the plugin is attached to.
```

#### Attaching plugins to multiple entities

> {{ site.kgo_product_name }} also supports to attach plugins to combination of multiple entities by `KongPluginBinding`.

Reviewer: ```suggestion
{{ site.kgo_product_name }} also supports attaching plugins to a combination of entities by `KongPluginBinding`.
```
"Combination" implies "multiple".

> Supported combinations includes:

Reviewer: ```suggestion
Supported combinations include:
```

* `Service` and `Route`
* `Service` and `Consumer`
* `Service` and `ConsumerGroup`

> * `Service`, `Route` and `Consumer`

Reviewer: ```suggestion
* `Service`, `Route`, and `Consumer`
```

> * `Service`, `Route` and `ConsumerGroup`

Reviewer: ```suggestion
* `Service`, `Route`, and `ConsumerGroup`
```
